app/src/main/java/com/example/final_project/database/WatchListRepository.java

```diff
@@ -26,6 +34,27 @@ public static synchronized WatchListRepository getRepository(Application applica
         return repository;
     }
 
+
+    public int insertMovie(Movie movie) {
+        try {
+            Future<Long> future = executor.submit(() -> movieDAO.insert(movie));
+            long id = future.get();
+            movie.setId((int) id);
+            return (int) id;
+        } catch (Exception e) {
+            e.printStackTrace();
+            return -1;
+        }
+    }
```

Comment:
Can this be reduced?
```suggestion
    public int insertMovie(Movie movie) {
        return (int) movieDAO.insert(movie)
    }
```